bindings/java/src/main/native/worker.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (C) Mellanox Technologies Ltd. 2001-2017.  ALL RIGHTS RESERVED.
+ * See file LICENSE for terms.
+ */
+#include "worker.h"
+
+ucs_status_t worker::init(ucp_worker_params_t params) {
```

Comment:
can this function be part of the constructor?

Reply:
It's easier to deal with errors this way - using return codes (ucs_status) instead of throwing exceptions. Do you prefer I'll move it into the c'tor?

Reply:
yes, ctor and exception is the c++ way